src/components/common/Form.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import Autocomplete from 'react-autocomplete';
+
+class Form extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedAddress: '',
+      isCollectionScheduleTextHidden: true,
+      isAutoTextHidden: false,
+      dateFormat: 'D/M/YYYY',
+      scheduleType: {
+        trash: 'trash',
+        recycle: 'recycle',
+        leaf: 'leaf',
+      },
+      dayOfWeek: {
+        sunday: 0,
+        monday: 1,
+        tuesday: 2,
+        wednesday: 3,
+        thrusday: 4,
+        friday: 5,
+        saturday: 6,
+      },
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  handleClick(event) {
+    this.setState({
+      isAutoTextHidden: false,
+      isCollectionScheduleTextHidden: true,
+      selectedAddress: '',
+    });
+  }
+  renderList(dataList) {
+    const selectedAddress = this.state.selectedAddress;
+    const items = dataList.map((item, index) => ({
+      id: item._id,
+      label: item.address1,
+    }));
+    return (
+      <div className="p-3 mb-2 bg-secondary text-white">
+        <div className="inner-addon right-addon">
+          <Autocomplete
+            getItemValue={item => item.label}
+            items={items}
+            renderItem={(item, isHighlighted) => (
+              <div key={item.id} style={{ background: isHighlighted ? 'lightgray' : 'black' }}>
+                {item.label}
+              </div>
+            )}
+            value={selectedAddress}
+            onChange={e =>
+              this.setState({
```

Comment:
it looks like you have indentical code here

```js
this.setState({
  selectedAddress: value,
  isCollectionScheduleTextHidden: false,
  isAutoTextHidden: true // this value differs
});
```

in cases like that you should refactor into a method and add a parameter for isAutoTextHidden

example (not saying to do it exactly like this, just to provide an example)

```js
const myFunction = isAutoTextHidden => {
  this.setState({
    selectedAddress: value,
    isCollectionScheduleTextHidden: false,
    isAutoTextHidden: isAutoTextHidden 
  });
};
```

Reply:
this function is also used in onSelect

Reply:
refactor into method to handle setState onChange() and onSelect()

Reply:
I had to modify code. there was an issue with my previous code.